src/util/GeoSparqlHelpers.cpp

```diff
@@ -54,6 +56,37 @@ double wktDistImpl(GeoPoint point1, GeoPoint point2) {
   return S2Earth::ToKm(S1Angle(p1, p2));
 }
 
+// ____________________________________________________________________________
+double kilometerToUnit(double kilometers,
+                       std::optional<UnitOfMeasurement> unit) {
+  double multiplicator = 1;
+  if (unit.has_value()) {
+    if (unit.value() == UnitOfMeasurement::METERS) {
+      multiplicator = 1000;
+    } else if (unit.value() == UnitOfMeasurement::KILOMETERS) {
+      multiplicator = 1;
+    } else if (unit.value() == UnitOfMeasurement::MILES) {
+      multiplicator = detail::kilometerToMile;
+    } else {
+      AD_CONTRACT_CHECK(unit.value() == UnitOfMeasurement::UNKNOWN);
+      AD_THROW("Unsupported unit of measurement for distance.");
```

Comment:
```suggestion
      AD_CORRECTNESS_CHECK(unit.value() == UnitOfMeasurement::UNKNOWN);
      AD_THROW("Unsupported unit of measurement for distance.");
```

+ add a test that throws in line 72.